op-e2e/actions/interop/interop_fork_test.go

```diff
@@ -0,0 +1,452 @@
+
+	// Run Isthmus until the interop upgrade
+	interopActivationOffset := 3
+	system := dsl.NewInteropDSL(t, dsl.SetInteropOffsetForAllL2s(uint64(interopActivationOffset)))
+
+	chainA := system.Actors.ChainA
+	chainB := system.Actors.ChainB
+	svr := system.Actors.Supervisor
+	chains := []*dsl.Chain{chainA, chainB}
+	expSyncStatuses := map[*dsl.Chain]*eth.SyncStatus{
+		chainA: {},
+		chainB: {},
+	}
+	var genesisTime uint64
+
+	////////////////////////////
+	// Before Interop Upgrade
+	////////////////////////////
+
+	for _, c := range chains {
+		expStatus := expSyncStatuses[c]
+		expStatus.UnsafeL2 = c.Sequencer.L2Unsafe()
+		expStatus.CrossUnsafeL2 = expStatus.UnsafeL2
+		expStatus.LocalSafeL2 = expStatus.UnsafeL2
+		expStatus.SafeL2 = expStatus.UnsafeL2
+		expStatus.FinalizedL2 = expStatus.UnsafeL2
+
+		// Sanity check: chain interop activation time was correctly configured
+		genesisTime = expStatus.UnsafeL2.Time
+		require.Equal(t, uint64(interopActivationOffset), *c.RollupCfg.InteropTime-genesisTime, "interop activation offset does not match configured expectation")
+
+		// Sanity check: chain is at genesis block and interop is not activated according to config and unsafe head timestamp
+		assertUnsafeTimeOffset(t, c, genesisTime, 0)
+		assertHeadBlockIDs(t, c, expStatus)
+		require.True(t, c.RollupCfg.IsIsthmus(expStatus.UnsafeL2.Time), "Isthmus should be active still in L2 chain with ID %d", c.ChainID)
+		require.False(t, c.RollupCfg.IsInterop(expStatus.UnsafeL2.Time),
+			"Interop should not be active yet in unsafe L2 chain with id %d. Time since genesis: %d, Interop activation offset: %d", c.ChainID, genesisTime-expStatus.UnsafeL2.Time, interopActivationOffset)
+
+		// Advance unsafe head by one
+		c.Sequencer.ActL2EmptyBlock(t)
+
+		// Interop still not active
+		assertUnsafeTimeOffset(t, c, genesisTime, 2)
+		updateExpectedUnsafe(t, c, expStatus, 1)
+		assertHeadBlockIDs(t, c, expStatus)
+		expStatus.UnsafeL2 = c.Sequencer.L2Unsafe()
+		require.True(t, c.RollupCfg.IsIsthmus(expStatus.UnsafeL2.Time), "Isthmus should be active still in L2 chain with ID %d", c.ChainID)
+		require.False(t, c.RollupCfg.IsInterop(expStatus.UnsafeL2.Time),
+			"Interop should not be active yet in unsafe L2 chain with id %d. Time since genesis: %d, Interop activation offset: %d", c.ChainID, genesisTime-expStatus.UnsafeL2.Time, interopActivationOffset)
+	}
+
+	// Settle both chains to DA
+	system.Actors.ActBatchAndMine(t, chains...)
+	assertL1Heads(t, system, 1, 0)
+
+	// Advance L1 safe head and finalized head
+	system.Actors.L1Miner.ActEmptyBlock(t)
+	system.Actors.L1Miner.ActL1SafeNext(t)
+	system.Actors.L1Miner.ActL1FinalizeNext(t)
+
+	// Verify
+	assertL1Heads(t, system, 2, 1)
+
+	// Still pre interop
+	for _, c := range chains {
+		expStatus := expSyncStatuses[c]
+
+		c.Sequencer.ActL2PipelineFull(t)
+
+		svr.SignalLatestL1(t)         // supervisor will be aware of latest L1
+		c.Sequencer.SyncSupervisor(t) // supervisor to react to exhaust-L1
```

Comment:
If I call `svr.ProcessFull(t)` on the next line the supervisor errors. This is mildly inconvenient, and is probably caused by some combination of the fact that the L2 genesis block has a 0x0 hash and that interop is not activated yet. @axelKingsley is this behavior defined at all?